utilities/transactions/write_unprepared_txn_db.cc

```diff
@@ -367,29 +365,49 @@ Iterator* WriteUnpreparedTxnDB::NewIterator(const ReadOptions& options,
   // TODO(lth): Refactor so that this logic is shared with WritePrepared.
   constexpr bool ALLOW_BLOB = true;
   constexpr bool ALLOW_REFRESH = true;
-  std::shared_ptr<ManagedSnapshot> own_snapshot = nullptr;
   SequenceNumber snapshot_seq;
   SequenceNumber min_uncommitted = 0;
-  if (options.snapshot != nullptr) {
-    snapshot_seq = options.snapshot->GetSequenceNumber();
-    min_uncommitted =
-        static_cast_with_check<const SnapshotImpl, const Snapshot>(
-            options.snapshot)
-            ->min_uncommitted_;
-  } else {
-    auto* snapshot = GetSnapshot();
-    // We take a snapshot to make sure that the related data in the commit map
-    // are not deleted.
-    snapshot_seq = snapshot->GetSequenceNumber();
-    min_uncommitted =
-        static_cast_with_check<const SnapshotImpl, const Snapshot>(snapshot)
-            ->min_uncommitted_;
-    own_snapshot = std::make_shared<ManagedSnapshot>(db_impl_, snapshot);
+  // TODO(lth): Currently, we can only create iterators with the transaction
+  // snapshot. This is because the iterator logic relies on snapshot validation
+  // to ensure that no committed keys exist between snapshot_seq < seq <
+  // max_visible_seq. This assumption is currently upheld by snapshot
+  // validation, and validation is run against the transaction snapshot sequence
+  // number.
+  //
+  // Therefore, we cannot guarantee correct results for iterators with no
+  // transaction snapshot, but with tracked keys. The second condition is so
```

Comment:
What if the snapshot was taken after the writes are performed? In that case the ValidateSnasphot logic has not prevented the scenario that might break the iterators. Perhaps we can a set a boolean anytime we do a write without snapshot, and check that boolean here to see if it safe to create iterators or not.

Reply:
I went instead of tracking the largest sequence number at which we did snapshot validation, in case snapshots were refreshed (which happens in MyRocks).